packages/vite-plugin-cloudflare/src/index.ts

```diff
@@ -407,67 +410,53 @@ export function cloudflare(pluginConfig: PluginConfig = {}): vite.Plugin[] {
 				// Skip this whole plugin if we are in preview mode
 				return !env.isPreview;
 			},
-			config() {
-				// Configure Vite with the Node.js polyfill aliases
-				// We have to do this across the whole Vite config because it is not possible to do it per Environment.
-				// These aliases are to virtual modules that then get Environment specific handling in the resolveId hook.
+			applyToEnvironment(environment) {
+				const workerConfig = getWorkerConfig(environment.name);
+				return isNodeCompat(workerConfig);
+			},
+			// We need the resolver from this plugin to run before built-in ones, otherwise Vite's built-in
+			// resolver will try to externalize the Node.js module imports (e.g. `perf_hooks` and `node:tty`)
+			// rather than allowing the resolve hook here to alias then to polyfills.
+			enforce: "pre",
+			configEnvironment() {
 				return {
```

Comment:
`configEnvironment` runs before `applyToEnvironment` so this configuration is currently being applied to all environments (including the default `client` and `ssr` environments).

```suggestion
			configEnvironment(name) {
				const workerConfig = getWorkerConfig(name);
				
				if (isNodeCompat(workerConfig)) {
				  return {
```

(Can't close the if statement in the suggestion because GitHub won't let me).

I would also move the `applyToEnvironment` hook after the `configEnvironment` hook to make this clearer.

Reply:
This is a great spot @jamesopstad - thanks. I didn't realise that the applyToEnvironments only affects which runtime hooks get called and not whether configureEnvironments is called.

Reply:
4de9f48d9c6af8eb2319b4cb8ce5f4f892b99c53